providers/openshift/provider.go

```diff
@@ -568,6 +568,44 @@ func (p *OpenShiftProvider) GetRedeemURL() (*url.URL, error) {
 	return redeemURL, err
 }
 
+func (p *OpenShiftProvider) DeleteSessionToken(session *providers.SessionState) error {
+	if len(session.AccessToken) == 0 {
+		return nil
+	}
+
+	// get an api path to the token
+	tokenAPIPath := getKubeAPIURLWithPath(fmt.Sprintf("/apis/oauth.openshift.io/v1/oauthaccesstokens/%s", session.AccessToken))
+
+	client, err := p.newOpenShiftClient()
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(http.MethodDelete, tokenAPIPath.String(), nil)
+	if err != nil {
+		log.Print("failed to build token deletion request")
+		return err
+	}
+
+	saToken, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	if err != nil || len(saToken) == 0 {
+		log.Printf("failed to read this workload's service account token: %v", err)
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", saToken))
```

Comment:
wrong token.  This needs to be the user token

Reply:
are we certain here?  does a user have auth to delete their own token?  Shouldn't it be the header has the token of the SA or the entity doing the request and from my PR the token to be deleted was in the url

Reply:
No, a token can always delete itself.  It may be possible that anyone can delete a token if they know the name, but @sttts has a design that may make that untrue in teh future.